### PR TITLE
Adding a rake tasks for seeding users and requests

### LINKED ISSUE #72: Create seed data
User might be tricky due to CAS. maybe we seed the users on the team.

### PR DESCRIPTION
**Pull request by carolyncole**
Created RandonRequestGenerator to randomly generate travel and absence requests
create make_me_a_supervisor to create additional fake users that you supervise
create make_me_a_department_head to create a fake department with fake supervisors who have fake reports
creates make_requests_for_everyone that generates random numbers of random request for every staff member in the system

fixes #72

### CODE REVIEW

File: db/structure.sql [hunk 1]
```diff
@@ -8,20 +8,6 @@ SET check_function_bodies = false;
 SET client_min_messages = warning;
 SET row_security = off;
 
---
```
Comment (hackmastera) [inline db/structure.sql]: sigh. our dbs do not agree.

File: lib/tasks/approvals.rake [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        status = ["pending", "approved", "denied"].sample
+        RandomRequestGenerator.generate_travel_request(creator: staff_profile, status: status)
+      end
+      1.upto(Random.rand(2...10)) do
+        status = ["pending", "approved", "denied"].sample
+        RandomRequestGenerator.generate_absence_request(creator: staff_profile, status: status)
+      end
+    end
+
+    puts "#{TravelRequest.count} TravelRequests generated and #{AbsenceRequest.count} AbsenceRequest generated for #{StaffProfile.count} Staff"
```
Comment (hackmastera) [inline lib/tasks/approvals.rake]: typo should be "AbsenceRequests"

File: spec/services/random_direct_reports_generator_spec.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
+require "rails_helper"
+
+RSpec.describe RandomDirectReportsGenerator, type: :model do
+  let(:supervisor) { FactoryBot.create :staff_profile }
+
+  before do
+    supervisor # call this here so we do not create the supervisor as part of the test
+  end
+
+  describe "##create_reports" do
```
Comment (hackmastera) [inline spec/services/random_direct_reports_generator_spec.rb]: what does the `##` mean?

File: db/structure.sql (hunk 1 above)
Comment (carolyncole) [reply db/structure.sql]: @hackmastera I will put your stuff back.  The comments do not cause me any issue.

File: spec/services/random_direct_reports_generator_spec.rb (hunk 3 above)
Comment (carolyncole) [reply spec/services/random_direct_reports_generator_spec.rb]: I believe ## means class method.

File: app/services/random_direct_reports_generator.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
+class RandomDirectReportsGenerator < ApplicationRecord
+  class << self
+    def create_reports(supervisor:, number_of_people: 5)
+      1.upto(number_of_people) do |_i|
+        create_staff(supervisor)
+      end
+    end
+
+    def create_department(head:, number_of_supervisors: 3)
```
[the remark this answers is not in the document]
Comment (hackmastera) [reply app/services/random_direct_reports_generator.rb]: @carolyncole that sounds great.

File: db/structure.sql (hunk 1 above)
Comment (hackmastera) [reply db/structure.sql]: 👍

File: spec/services/random_direct_reports_generator_spec.rb (hunk 3 above)
Comment (hackmastera) [reply spec/services/random_direct_reports_generator_spec.rb]: ah, okay. I usually see `.` for class method? but 👍 this works now that I know 😄

Comment (carolyncole) [reply spec/services/random_direct_reports_generator_spec.rb]: I'm not sure what convention I was following, but I am happy to follow the local convention.  I changed them to `.`.

**Pull request merged**